src/backend/distributed/utils/maintenanced.c

```diff
@@ -345,6 +366,42 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 
 
 /*
+ * MetadataTableAccessible returns whether the given metadata table exists
+ * and can be locked using an AccessShareLock.
+ */
+static bool
+MetadataTableAccessible(const char *tableName)
+{
+	bool tableAccessible = false;
+
+	Oid relationId = get_relname_relid(tableName, PG_CATALOG_NAMESPACE);
```

Comment:
What do you mean by trylock callback?

We tried `RangeVarGetRelidExtended` at first, but realised it would throw an error if it could not obtain a lock. We then tried putting it in PG_TRY, but that turned out to be a bad idea in a background worker.